codec-http/src/test/java/io/netty/handler/codec/http/HttpRequestDecoderTest.java

```diff
@@ -639,6 +639,34 @@ void headerValuesMayBeBracketedByZeroOrMoreWhitespace() throws Exception {
         assertFalse(channel.finish());
     }
 
+    @Test
+    public void testChunkSizeOverflow() {
+        String requestStr = "PUT /some/path HTTP/1.1\r\n" +
+                "Transfer-Encoding: chunked\r\n\r\n" +
+                "8ccccccc\r\n";
+        EmbeddedChannel channel = new EmbeddedChannel(new HttpRequestDecoder());
+        assertTrue(channel.writeInbound(Unpooled.copiedBuffer(requestStr, CharsetUtil.US_ASCII)));
+        HttpRequest request = channel.readInbound();
+        assertTrue(request.decoderResult().isSuccess());
+        HttpContent c = channel.readInbound();
+        c.release();
```

Comment:
1. In both tests, consider asserting that `c.decoderResult().cause()` is `NumberFormatException`.
2. Consider adding similar tests for `HttpResponseDecoderTest`.